src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,124 @@
+    public static final Comparator<Person> COMPARE_BY_ID =
+            Comparator.comparing(person -> person.getStudentId().toString());
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Sorts contacts according to 1 specified field and order. \n"
+            + "Parameters: "
+            + "ORDER (must be 1 (ascending) or -1 (descending)) "
+            + "[" + PREFIX_NAME + "] "
+            + "[" + PREFIX_STUDENT_ID + "] \n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME;
+
+    private final Comparator<Person> sortingComparator;
+    private final boolean isAscendingOrder;
+
+    /**
+     * Constructs a {@code SortCommand} with the given comparator and order.
+     *
+     * @param personComparator Comparator to sort persons.
+     * @param isAscendingOrder {@code true} for ascending, {@code false} for descending.
+     * @throws NullPointerException if {@code personComparator} is null.
+     */
+    private SortCommand(Comparator<Person> personComparator, boolean isAscendingOrder) {
+        requireNonNull(personComparator);
+        this.sortingComparator = personComparator;
+        this.isAscendingOrder = isAscendingOrder;
+    }
+
+    /**
+     * Returns a {@code SortCommand} to sort by name.
+     *
+     * @param order 1 for ascending, -1 for descending.
+     */
+    public static SortCommand sortByName(Integer order) {
+        requireNonNull(order);
+        boolean isAscendingOrder = (order == 1);
+        return new SortCommand(COMPARE_BY_NAME, isAscendingOrder);
+    }
+
+    /**
+     * Returns a {@code SortCommand} to sort by student ID.
+     *
+     * @param order 1 for ascending, -1 for descending.
+     */
+    public static SortCommand sortByStudentId(Integer order) {
+        requireNonNull(order);
+        boolean isAscendingOrder = (order == 1);
+        return new SortCommand(COMPARE_BY_ID, isAscendingOrder);
+    }
+
+    /**
+     * Returns a success message indicating the sorting field and order.
+     *
+     * @return A message describing the sorting field and order, or an empty string if unknown.
+     */
+    public String getMessageSuccess() {
+        String order = isAscendingOrder ? "Ascending" : "Descending";
+        if (this.sortingComparator == COMPARE_BY_NAME) {
+            return "Sorted by Name in " + order + " order.";
+        } else if (this.sortingComparator == COMPARE_BY_ID) {
+            return "Sorted by Student Id in " + order + " order.";
+        } else {
+            return "";
+        }
+    }
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        Comparator<Person> personComparator =
+                isAscendingOrder ? this.sortingComparator : this.sortingComparator.reversed();
+        model.updateSortedPersonList(personComparator);
+        return new CommandResult(getMessageSuccess());
+    }
```

Comment:
Just nitpicking but not too sure if there should be command exception thrown if the lis of people is null?

Reply:
I kept the logic consistent with ListCommand, perhaps we could put file this as a bug and resolve this in both commands together?